### DISCUSSION TITLE
How can I modify code location <-> daemon / webserver timeouts?

### DISCUSSION THREAD

**Discussion by tomasz-maj-bp** [dagster-io/dagster#22587] [Q&A]
Hey, I'm trying to load code location that sometimes takes more then 60s to do so. I've noticed that during those times I'm encountering some timeout breaches and from code location perspective I can see:
> INFO:dagster.code_server:Stopping server once all current RPC calls terminate or 600 seconds pass

from webserver:
> INFO:dagster-webserver:Received LocationStateChangeEventType.LOCATION_ERROR event for location stresstest-volume, refreshing

and some generic error in daemon:
> status = StatusCode.UNAVAILABLE
details = "Received http2 header with status: 503"

I've tried to change some environment variables:
> DAGSTER_GRPC_TIMEOUT_SECONDS = 600
DAGSTER_REPOSITORY_GRPC_TIMEOUT_SECONDS = 600

and add extra yaml values:
> additionalInstanceConfig:
  code_servers:
    reload_timeout: 600
    local_startup_timeout: 600
    
 But neither seems to affect it. I'm using dagster 1.7.2.
 I recall that there were env variables in earlier version for it, but I can't find them now.

**Comment by garethbrickman** [comment]
Increasing the timeout in your `dagster.yaml` this way is recommended in this issue https://github.com/dagster-io/dagster/issues/8561

```
code_servers:
  local_startup_timeout: 120
```

**Reply by tomasz-maj-bp** [reply]
Hey Gareth, thanks for your reply. As explained in my main post, I've already tried to add that parameter in https://github.com/dagster-io/dagster/blob/master/helm/dagster/values.yaml#L1346 (and deploy) but it haven't had any effect.

**Reply by garethbrickman** [reply]
If using a Kubernetes agent configured with Helm chart, ­[in the chart template](https://artifacthub.io/packages/helm/dagster-cloud/dagster-cloud-agent?modal=values) there is:

`serverProcessStartupTimeout`​​ - Specifies the timeout in seconds the agent should use when waiting for a code server to be ready after the Kubernetes deployment is created. This timeout should be increased if your code locations can take more than 3 minutes to load your Dagster definitions.

This is the time required for the code location container to start to the grpc server being ready and serving request. If you are using many libraries, or are creating a dbt manifest dynamically, you may need to increase this.

​`deploymentStartupTimeout​​` - Specifies the timeout in seconds when creating a new Kubernetes deployment for a code server. If not set, defaults to 300 seconds.

This is the time the agent will wait from the deployment initialization to the pod being running. If your cluster depends on slow node provisioning when starting new pods, you might need to increase that.
